docs/adr/2022-07-22_019-reference-scripts.md

```diff
@@ -0,0 +1,61 @@
+---
+slug: 19
+title: | 
+  19. Use of reference scripts
+authors: []
+tags: [Proposed]
+---
+
+## Status
+
+Proposed
+
+## Context
+
+* In the desire to make Hydra transactions smaller and cheaper (at the time of writing any abort tx was too big), we want to use the **reference script** and **reference input** features of the upcoming `Babbage` ledger era. See the [babbage ledger spec](https://hydra.iohk.io/build/16861604/download/1/babbage-changes.pdf), [CIP-31](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0031) and [CIP-33](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0033) for details.
+
+* With these features we not need to (re-)include the same scripts in each transaction.
+
+* The CIPs do not specify how reference scripts are to be managed and we can see at least two options:
+  1. Add them as outputs to the `init` transaction or prior that as part of each Hydra Head instance
+  2. Post them out-of-band, separate to individual Head instances
+
+* Ownership of the outputs holding the scripts is to be considered. If these "reference outputs" are spent, they cannot be referred to anymore. This would mean all heads referring ot them become denialed of service (DoS).
```

Comment:
```suggestion
* Ownership of the outputs holding the scripts is to be considered. If these "reference outputs" are spent, they cannot be referred to anymore. This would mean all heads referring to them can be denied of service (DoS).
```